### PR TITLE
Add esql qa resources to conf.yaml

### PR DESCRIPTION
**Pull request by ChrisHegarty**
This commit adds the x-pack/plugin/esql/qa/testFixtures/src/main/resources resource for building the Elasticsearch. The directory contains files that are used for code snippet testing.

Currently only required for the `feature/esql` branch, but will be required for `main` once merged, post 8.10.

### CODE REVIEW

File: conf.yaml [hunk 1]
```diff
@@ -347,6 +347,11 @@ contents:
                 path:   x-pack/qa/sql
                 # only exists from 6.3 to 6.5
                 exclude_branches:   [ main, 8.9, 8.8, 8.7, 8.6, 8.5, 8.4, 8.3, 8.2, 8.1, 8.0, 7.17, 7.16, 7.15, 7.14, 7.13, 7.12, 7.11, 7.10, 7.9, 7.8, 7.7, 7.6, 7.5, 7.4, 7.3, 7.2, 7.1, 7.0, 6.8, 6.7, 6.6, 6.2, 6.1, 6.0, 5.6, 5.5, 5.4, 5.3, 5.2, 5.1, 5.0, 2.4, 2.3, 2.2, 2.1, 2.0, 1.7, 1.6, 1.5, 1.4, 1.3, 0.90 ]
+              -
+                repo:   elasticsearch
+                path:   x-pack/plugin/esql/qa/testFixtures/src/main/resources
+                # only exists from 8.11  (and the feature/esql branch)
+                exclude_branches:   [ main, 8.10, 8.9, 8.8, 8.7, 8.6, 8.5, 8.4, 8.3, 8.2, 8.1, 8.0, 7.17, 7.16, 7.15, 7.14, 7.13, 7.12, 7.11, 7.10, 7.9, 7.8, 7.7, 7.6, 7.5, 7.4, 7.3, 7.2, 7.1, 7.0, 6.8, 6.7, 6.6, 6.2, 6.1, 6.0, 5.6, 5.5, 5.4, 5.3, 5.2, 5.1, 5.0, 2.4, 2.3, 2.2, 2.1, 2.0, 1.7, 1.6, 1.5, 1.4, 1.3, 0.90 ]
```
Comment (abdonpijpelink) [inline conf.yaml:354]:
```suggestion
                exclude_branches:   [ main, 8.10, 8.9, 8.8, 8.7, 8.6, 8.5, 8.4, 8.3, 8.2, 8.1, 8.0, 7.17, 7.16, 7.15, 7.14, 7.13, 7.12, 7.11, 7.10, 7.9, 7.8, 7.7, 7.6, 7.5, 7.4, 7.3, 7.2, 7.1, 7.0, 6.8, 6.7, 6.6, 6.5, 6.4, 6.3, 6.2, 6.1, 6.0, 5.6, 5.5, 5.4, 5.3, 5.2, 5.1, 5.0, 2.4, 2.3, 2.2, 2.1, 2.0, 1.7, 1.6, 1.5, 1.4, 1.3, 0.90 ]
```

Comment (abdonpijpelink) [reply conf.yaml:354]: CI is failing because of some missing intermediate versions.

**Review by abdonpijpelink — APPROVED** [review]
LGTM

**Comment by nik9000** [comment]
I'm just getting to this. It looks right. Welcome to the giant yaml file that runs all the docs!

**Pull request merged**